src/feed_builder.rs

```diff
@@ -29,11 +30,28 @@ where
 {
   /// Create a new instance.
   #[inline]
-  pub fn new(public_key: PublicKey, storage: Storage<T>) -> Self {
-    Self {
-      storage,
-      public_key,
-      secret_key: None,
+  pub fn new(public_key: Option<PublicKey>, mut storage: Storage<T>) -> Self {
```

Comment:
I'm really not sure if we'd want the logic here or directly in the Feed.  It also feels weird to have an `Option` as first argument but I wanted to avoid a BC break.

Reply:
Backward compatibility break, but I then realized that I had change the `public_key` signature so my code is breaking the backward compatibility promise. Anyway I think that it's a bit early into the project to care about such things :sweat_smile:.